Rubberduck.VBEEditor/ComManagement/TypeLibs/TypeInfos.cs

```diff
@@ -718,18 +718,19 @@ public TypeInfoWrapper GetControlType(string controlName)
 
         public TypeInfoWrapper GetSafeRefTypeInfo(int hRef)
         {
-            IntPtr typeInfoPtr = IntPtr.Zero;
             // we cast to our ITypeInfo_Ptrs interface in order to work with the raw IntPtr for aggregation
-            ((ITypeInfo_Ptrs)target_ITypeInfo).GetRefTypeInfo(hRef, out typeInfoPtr);
+            ((ITypeInfo_Ptrs)target_ITypeInfo).GetRefTypeInfo(hRef, out var typeInfoPtr);
             var outVal = new TypeInfoWrapper(typeInfoPtr, _isUserFormBaseClass ? (int?)hRef : null); // takes ownership of the COM reference
+
+            _typeInfosWrapped?.Dispose();
             _typeInfosWrapped = _typeInfosWrapped ?? new DisposableList<TypeInfoWrapper>();
```

Comment:
this looks weird. The list is disposed and then reused? That makes no sense.

Reply:
I'm sure those three items got flagged because the analyzer thought those would be potentially reassigned. Reassignment without disposing the prior value would be problematic. cf. [IDISP003](https://github.com/DotNetAnalyzers/IDisposableAnalyzers/blob/master/documentation/IDISP003.md)